trajectory_utils/src/trajectory_utils/conversions/conversions.cpp

```diff
@@ -120,7 +120,33 @@ void time_to_speed(const std::vector<double>& downtrack, const std::vector<doubl
     double dt = cur_time - prev_time;
     double delta_d = cur_pos - prev_position;
 
-    double cur_speed = (2.0 * delta_d / dt) - prev_speed;
+    double cur_speed;
+    double jerk_min = 0.01; //Min stop and wait jerk
+    
+    //if stop and wait, use constant jerk equations
+    if(isStopandWait[i]){
+
+      if(delta_d == 0){
+        cur_speed =0;
+        break;
```

Comment:
This break is a bit problematic since the function can now return an unequal lenth vector. This is the core contract of this method that the output vector has the same size. In this case I would think storing a cur_speed of 0 and moving to the next point would be preferable to breaking.

Reply:
If this case can truly not be handled then throwing an exception would be appropriate.